# **Inteligencia Artificial con Aplicaciones en Economía I**

- 👩‍🏫 **Profesora:** [Lina María Castro](https://www.linkedin.com/in/lina-maria-castro)  
- 📧 **Email:** [lmcastroco@gmail.com](mailto:lmcastroco@gmail.com)  
- 🎓 **Universidad:** Universidad Externado de Colombia - Facultad de Economía

# **Taller Limpieza y transformación de datos**🧹

**IMPORTANTE**: Guarda una copia de este notebook en tu Google Drive o computador.

**Taller en grupos de 3 o parejas.**

**Nombres estudiantes:**

-Juan Esteban Barrantes Riaño
-
-

**Forma de entrega:**

- Nombrar el archivo de la siguiente forma:“Taller_Limpieza_nombres_y_appellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.office.com/r/6GZ89uiLNC.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

**Plazo de entrega:**

19 de marzo de 2026, máximo a las 11:59 p.m. Tenga en cuenta que luego de esa hora el formulario en forms se cierra. El Jupupyter Notebook también debe quedar subido en Github antes de esa hora.

**Instrucciones Generales:**

Completa el código en las celdas marcadas con `### TU CÓDIGO AQUÍ ###`. Puedes añadir más celdas si lo requieres.

Utilice la base de la GEIH de julio a diciembre de 2025, la cual se encuentra en https://drive.google.com/drive/folders/1Th4TWdsOahAAvBUKxixFhaELC_KPsl3y?usp=sharing. Tenga en cuenta sep="," y encoding='latin1'. Las variables 'DPTO' y 'RAMA4D_R4' deben ser cargadas como string.

# Ejercicio 1: Importar librerías y cargar datos

1.	Importe todas las librerías que vaya a utilizar.

In [78]:
### TU CÓDIGO AQUÍ ###

import os
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

2.	Mejore la visualización de los dataframes como se ha visto en clase.

In [79]:
### TU CÓDIGO AQUÍ ###
pd.options.display.max_columns = None
pd.options.display.float_format = '{:,.2f}'.format

3.	Conéctese a su Google Drive y establezca la ruta del dataset.

In [80]:
### TU CÓDIGO AQUÍ ###
from google.colab import drive, files
drive.mount('/content/drive')


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [81]:
path = '/content/drive/MyDrive/datasets'

In [82]:
os.chdir(path)

4.	Cargue los datos en el notebook. Llame al dataframe 'df_geih'

In [83]:
### TU CÓDIGO AQUÍ ###
df = pd.read_csv('GEIH_2025_Jul_Dic_Consolidado.csv', sep=',', encoding='latin1', dtype={'DPTO': str, 'RAMA4D_R4': str})

In [84]:
df

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,FEX_C18,DPTO,PT,P6016,P3271,P6040,P6030S1,P6030S3,P6050,P6083,P6083S1,P6081,P6081S1,P2057,P2059,P2061,P6080,P6080S1,P6080S1A1,P6070,P6071,P6071S1,P6090,P6100,P6110,P6120,P1906S1,P1906S2,P1906S3,P1906S4,P1906S5,P1906S6,P1906S7,P1906S8,P6160,P6170,P3041,P3042,P3042S1,P3042S2,P3043,P3043S1,P3038,P3039,POB_MAY18,P4005,P4010,P4020,P4030S1,P4030S1A1,P4030S2,P4030S3,P4030S4,P4030S4A1,P4030S5,P5000,P5010,P5020,P5030,P5040,P5050,P5070,P5080,P5090,P5090S1,P5100,P5110,P5130,P5140,P5222S1,P5222S10,P5222S11,P5222S2,P5222S3,P5222S4,P5222S5,P5222S6,P5222S7,P5222S8,P5222S8A1,P5222S9,P6008,P70,FFT,FT,P3362S1,P3362S2,P3362S3,P3362S4,P3362S5,P3362S6,P3362S7,P3362S7A1,P3362S8,P6240,P6240S1,P6240S2,P6250,P6260,P6260S1,P6260S1A1,P6260S2,P6270,P6280,P6300,P6310,P6310S1,P6320,P6330,P6340,P6350,P6351,PET,INGLABO,OCI,OFICIO_C8,P1800,P1800S1,P1801S1,P1801S2,P1801S3,P1802,P1805,P1879,P1880,P1880S1,P1881,P1881S1,P1882,P3044S2,P3045S1,P3045S2,P3045S3,P3046,P3047,P3048,P3049,P3051,P3052,P3053,P3054,P3054S1,P3055,P3055S1,P3056,P3057,P3058S1,P3058S2,P3058S3,P3058S4,P3058S5,P3059,P3061,P3062S1,P3062S2,P3062S3,P3062S4,P3062S5,P3062S6,P3062S7,P3062S8,P3062S9,P3063,P3063S1,P3064,P3064S1,P3065,P3066,P3067,P3067S1,P3067S2,P3068,P3069,P3071S3,P3072S2,P3073,P3363,P3364,P3364S1,P3365,P3365S1,P514,P515,P550,P6400,P6410,P6420S2,P6422,P6424S1,P6424S2,P6424S3,P6424S5,P6426,P6430,P6430S1,P6440,P6450,P6460,P6460S1,P6500,P6510,P6510S1,P6510S2,P6545,P6545S1,P6545S2,P6580,P6580S1,P6580S2,P6585S1,P6585S1A1,P6585S1A2,P6585S2,P6585S2A1,P6585S2A2,P6585S3,P6585S3A1,P6585S3A2,P6585S4,P6585S4A1,P6585S4A2,P6590,P6590S1,P6600,P6600S1,P6610,P6610S1,P6620,P6620S1,P6630S1,P6630S1A1,P6630S2,P6630S2A1,P6630S3,P6630S3A1,P6630S4,P6630S4A1,P6630S6,P6630S6A1,P6640,P6640S1,P6750,P6760,P6765,P6765S1,P6775,P6780,P6780S1,P6790,P6800,P6810,P6810S1,P6830,P6830S1,P6850,P6880,P6880S1,P6915,P6915S1,P6920,P6930,P6940,P6960,P6990,P7020,P7026,P7028,P7028S1,P7040,P7045,P7050,P7070,P7075,P7077,P7090,P7100,P7110,P7120,P7130,P7140S1,P7140S2,P7140S3,P7140S4,P7140S5,P7140S6,P7140S7,P7140S8,P7140S9,P7140S9A1,P7150,P7160,P7170S1,P7170S5,P7170S6,P7180,P7240,P7240S1,P760,P9440,P9450,RAMA2D_R4,RAMA4D_R4,DESCRIPCION_CIIU,NOMBRE_DPTO
0,20250731,7,2025,8283143,1,1,1,10,NaN,2,27.75,08,1,1,1,72,8.00,"1,952.00",1,3,NaN,3,NaN,1,NaN,1,6,NaN,NaN,2.00,1.00,2.00,1,3.00,NaN,NaN,4,4,4,3,3,4,4,4,2.00,2.00,NaN,1.00,0.00,NaN,NaN,NaN,2.00,1.00,1.00,1,5,1,1,1.00,2,2,2,NaN,2,2,1,6,NaN,4,6,5,5.00,4,NaN,NaN,NaN,"200,000.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,2,1,NaN,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,NaN,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,"790,000.00",1.00,"9,212.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,3.00,NaN,3.00,NaN,10.00,143.00,2.00,2.00,2.00,2.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2.00,NaN,NaN,NaN,1.00,2.00,NaN,NaN,NaN,1.00,1.00,NaN,1.00,NaN,NaN,1.00,2.00,2.00,2.00,2.00,84.00,7.00,NaN,1.00,1.00,NaN,NaN,"590,000.00",2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,NaN,2.00,NaN,1.00,"200,000.00",2.00,NaN,2.00,NaN,2.00,NaN,2.00,NaN,2.00,NaN,2.00,NaN,2.00,NaN,2.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,12.00,40.00,NaN,NaN,NaN,NaN,40.00,8.00,NaN,2.00,NaN,2.00,NaN,NaN,NaN,2.00,1.00,3.00,4.00,NaN,2.00,NaN,NaN,NaN,NaN,NaN,2.00,NaN,NaN,NaN,2.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,1.00,1.00,2.00,3.00,NaN,0.00,2.00,2.00,1.00,0143,Cría de ovejas y cabras,ATLÁNTICO
1,20250731,7,2025,8283143,1,2,1,10,NaN,2,27.75,08,1,2,2,34,8.00,"1,990.00",2,3,NaN,3,NaN,1,NaN,1,6,NaN,NaN,2.00,1.00,1.00,2,NaN,NaN,NaN,4,4,4,4,4,4,4,4,2.00,2.00,NaN,1.00,0.00,NaN,NaN,NaN,1.00,2.00,1.00,1,5,1,1,1.00,2,2,2,NaN,2,2,1,6,NaN,4,6,5,5.00,4,NaN,NaN,NaN,"200,000.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,2,1,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.00,NaN,NaN,2.00,2.

5. Obtenga las dimensiones de la base

In [85]:
### TU CÓDIGO AQUÍ ###
df.shape

(409127, 314)

# Ejercicio 2:  Seleccionar y renombrar columnas

La GEIH contiene cientos de variables. Para nuestro análisis del mercado laboral, nos enfocaremos en un subconjunto.

Adicionalmente, los nombres de las columnas en la GEIH son códigos. Para hacer el análisis más intuitivo, vamos a renombrarlas.

1. Crea un nuevo DataFrame llamado `df_empleo` que contenga únicamente las columnas que están en la lista `columnas_interes`.

In [86]:
columnas_interes = ['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'MES', 'CLASE', 'DPTO', 'FEX_C18', 'PT', 'P3271', 'P6040', 'P6080', 'P3043', 'OCI',
                    'P6430', 'RAMA4D_R4', 'INGLABO', 'DESCRIPCION_CIIU', 'NOMBRE_DPTO']

In [87]:
### TU CÓDIGO AQUÍ ###
df_empleo = df[columnas_interes]

2. Renombra las columnas del DataFrame `df_empleo` utilizando el diccionario `mapa_nombres`.

In [88]:
mapa_nombres = {
'DIRECTORIO': 'id_vivienda',
'SECUENCIA_P': 'id_hogar',
'ORDEN': 'id_persona',
'MES': 'mes',
'CLASE': 'clase',
'DPTO': 'departamento',
'FEX_C18': 'factor_expansion',
'PT': 'poblacion_total',
'P3271': 'sexo',
'P6040': 'edad',
'P6080': 'grupo_etnico',
'P3043': 'nivel_educativo',
'OCI': 'ocupados',
'P6430': 'categoria_ocupacional',
'RAMA4D_R4': 'cod_ciiu',
'INGLABO': 'ingreso_laboral',
'DESCRIPCION_CIIU': 'desc_ciiu',
'NOMBRE_DPTO': 'nombre_dpto'}

In [89]:
### TU CÓDIGO AQUÍ ###
df_empleo.rename(columns=mapa_nombres, inplace=True)

3. Obtenga las dimensiones de la base

In [90]:
### TU CÓDIGO AQUÍ ###
df_empleo.shape

(409127, 18)

# Ejercicio 3: Revisión de duplicados

Un paso crucial en la limpieza es asegurar que no haya observaciones duplicadas que puedan sesgar nuestro análisis.

1. Verifica si hay filas duplicadas en `df_empleo`.

In [91]:
### TU CÓDIGO AQUÍ ###
df_empleo[df_empleo.duplicated()]

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto


2. Cuenta cuántas filas duplicadas existen.

In [92]:
### TU CÓDIGO AQUÍ ###
df_empleo.duplicated().sum()

np.int64(0)

3. Elimina las filas duplicadas, si las hay.

In [93]:
### TU CÓDIGO AQUÍ ###
#no hay filas duplicadas

4. Verifica si hay filas duplicadas específicamente en las columnas 'id_vivienda', 'id_hogar', 'id_persona'.

In [94]:
### TU CÓDIGO AQUÍ ###
df_empleo[df_empleo.duplicated(subset=["id_vivienda","id_hogar","id_persona"], keep=False)]

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto


# Ejercicio 4: Valores faltantes

Los datos del mundo real raramente están completos. Aquí, trataremos los valores faltantes en la columna de ingresos.


1. Calcula el número de valores nulos (`NaN`) en la columna `ingreso_laboral` **antes** de cualquier modificación.

In [95]:
### TU CÓDIGO AQUÍ ###
len(df_empleo[df_empleo['ingreso_laboral'].isna()])

231144

2. En la GEIH, un ingreso faltante a menudo significa que la persona no reportó ingresos o no aplica (ej. no trabaja). Una estrategia común es imputar estos valores con `0`. Rellena los valores `NaN` en la columna `ingreso_laboral` con `0`.

In [96]:
### TU CÓDIGO AQUÍ ###
df_empleo['ingreso_laboral'] = df_empleo['ingreso_laboral'].fillna(0)
df_empleo.tail()

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
409122,8473907,1,1,12,2,73,378.31,1,2,61,6,NaN,1.00,3.00,9700,"1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,1,54,534.55,1,1,58,6,9.00,1.00,2.00,8530,"3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,1,54,534.55,1,2,51,6,9.00,1.00,2.00,8412,"7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,1,54,534.55,1,2,31,6,7.00,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER
409126,8473908,1,4,12,1,54,534.55,1,2,26,6,7.00,1.00,4.00,4791,"500,000.00",Comercio al por menor realizado a través de in...,NORTE DE SANTANDER


In [97]:
len(df_empleo[df_empleo['ingreso_laboral'].isna()])

0

# Ejercicio 5: Decodificar variables



Las variables categóricas a menudo necesitan ser "traducidas" para hacerlas más entendibles para el análisis.

1. Usa los siguientes diccionarios para reemplazar los códigos numéricos por etiquetas de texto en las siguientes columnas: 'clase', 'sexo', 'grupo_etnico', 'nivel_educativo' y 'categoria_ocupacional'. Muestra los conteos de las nuevas categorías usando `.value_counts()` para verificar el cambio en cada una de las columnas.

In [98]:
dic_clase = {1: 'Cabeceras', 2: 'Centros poblados y rural disperso'}
dic_sexo = {1: 'Hombre', 2: 'Mujer'}
dic_etnico = {1: 'Indígena', 2: 'Gitano (Rom)', 3: 'Raizal del archipiélago', 4: 'Palenquero',
              5: 'Negro, mulato (afrodescendiente)', 6: 'Ninguno de los anteriores'}
dic_nivel_educativo = {1: 'Ninguno', 2: 'Educación media', 3: 'Educación media',
                       4: 'Educación media', 5: 'Técnica profesional', 6: 'Tecnológica',
                       7: 'Universitaria', 8: 'Especialización', 9: 'Maestría',
                       10: 'Doctorado', 99: 'No sabe, no informa'}
dic_cat_ocu = {1: 'Asalariado', 2: 'Asalariado', 3: 'Asalariado',
               8: 'Asalariado', 4: 'Independiente', 5: 'Independiente',
               9: 'Independiente', 6: 'Trabajador sin remuneración',
               7: 'Trabajador sin remuneración'}

In [99]:
### TU CÓDIGO AQUÍ ###
df_empleo['clase'] = df_empleo['clase'].replace(dic_clase)
df_empleo['clase'].value_counts()

,count
clase,
Cabeceras,356377
Centros poblados y rural disperso,52750


In [100]:
### TU CÓDIGO AQUÍ ###
df_empleo['sexo'] = df_empleo['sexo'].replace(dic_sexo)
df_empleo['sexo'].value_counts()

,count
sexo,
Mujer,218597
Hombre,190530


In [101]:
### TU CÓDIGO AQUÍ ###
df_empleo['grupo_etnico'] = df_empleo['grupo_etnico'].replace(dic_etnico)
df_empleo['grupo_etnico'].value_counts()

,count
grupo_etnico,
Ninguno de los anteriores,350562
"Negro, mulato (afrodescendiente)",39032
Indígena,18420
Raizal del archipiélago,925
Palenquero,120
Gitano (Rom),68


In [102]:
### TU CÓDIGO AQUÍ ###
df_empleo['nivel_educativo'] = df_empleo['nivel_educativo'].replace(dic_nivel_educativo)
df_empleo['nivel_educativo'].value_counts()

,count
nivel_educativo,
Educación media,116635
Universitaria,32517
Técnica profesional,24503
Ninguno,12963
Tecnológica,10640
Especialización,7927
Maestría,3154
Doctorado,339
"No sabe, no informa",1


In [103]:
### TU CÓDIGO AQUÍ ###
df_empleo['categoria_ocupacional'] = df_empleo['categoria_ocupacional'].replace(dic_cat_ocu)
df_empleo['categoria_ocupacional'].value_counts()

,count
categoria_ocupacional,
Asalariado,92016
Independiente,82498
Trabajador sin remuneración,7339


# Ejercicio 6: Reorganizar columnas

1. Reorganiza las columnas con el siguiente orden: 'id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto',     'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral'

In [104]:
### TU CÓDIGO AQUÍ ###
df_empleo.columns

Index(['id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento',
       'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico',
       'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu',
       'ingreso_laboral', 'desc_ciiu', 'nombre_dpto'],
      dtype='object')

In [105]:
df_empleo = df_empleo[['id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto', 'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral']]
df_empleo.head()

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
0,8283143,1,1,7,Centros poblados y rural disperso,08,ATLÁNTICO,27.75,1,Hombre,72,Ninguno de los anteriores,NaN,1.00,Trabajador sin remuneración,0143,Cría de ovejas y cabras,"790,000.00"
1,8283143,1,2,7,Centros poblados y rural disperso,08,ATLÁNTICO,27.75,1,Mujer,34,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
2,8283145,1,1,7,Centros poblados y rural disperso,08,ATLÁNTICO,59.11,1,Hombre,45,Ninguno de los anteriores,NaN,1.00,Asalariado,0141,Cría de ganado bovino y bufalino,"1,100,000.00"
3,8283145,1,2,7,Centros poblados y rural disperso,08,ATLÁNTICO,59.11,1,Mujer,50,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
4,8283145,1,3,7,Centros poblados y rural disperso,08,ATLÁNTICO,59.11,1,Hombre,19,Ninguno de los anteriores,NaN,1.00,Trabajador sin remuneración,0141,Cría de ganado bovino y bufalino,"820,000.00"


# Ejercicio 7: Cambiar tipo de dato


Para realizar cálculos y análisis correctamente, las columnas deben tener el tipo de dato adecuado (numérico, categórico, etc.).

1.  Muestra en pantalla los tipos de datos actuales de `df_empleo`.

In [106]:
### TU CÓDIGO AQUÍ ###
df_empleo.dtypes

,0
id_vivienda,int64
id_hogar,int64
id_persona,int64
mes,int64
clase,object
departamento,object
nombre_dpto,object
factor_expansion,float64
poblacion_total,int64
sexo,object


2.  Las siguientes columnas representan categorías: 'clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu'. Conviértelas al tipo `category` para optimizar el uso de memoria y facilitar análisis futuros.

In [107]:
### TU CÓDIGO AQUÍ ###
categorias = ['clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu']
for categoria in categorias:
  df_empleo[categoria] = df_empleo[categoria].astype('category')
df_empleo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 409127 entries, 0 to 409126
Data columns (total 18 columns):
 #   Column                 Non-Null Count   Dtype   
---  ------                 --------------   -----   
 0   id_vivienda            409127 non-null  int64   
 1   id_hogar               409127 non-null  int64   
 2   id_persona             409127 non-null  int64   
 3   mes                    409127 non-null  int64   
 4   clase                  409127 non-null  category
 5   departamento           409127 non-null  category
 6   nombre_dpto            409127 non-null  category
 7   factor_expansion       409127 non-null  float64 
 8   poblacion_total        409127 non-null  int64   
 9   sexo                   409127 non-null  category
 10  edad                   409127 non-null  int64   
 11  grupo_etnico           409127 non-null  category
 12  nivel_educativo        208679 non-null  category
 13  ocupados               181853 non-null  float64 
 14  categoria_ocupaciona

# Ejercicio 8: Ordenar el dataframe (5%)

1. Ordena el dataframe por la columna "edad" de mayor a menor. Observa la columna 'edad' y escribe en una celda de markdown cuál fue la edad mínima y la edad máxima de las personas que respondieron la encuesta de la GEIH.

In [108]:
### TU CÓDIGO AQUÍ ###
df_empleo.sort_values(by='edad', ascending = False)

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
256896,8395695,1,1,10,Cabeceras,18,CAQUETÁ,40.61,1,Mujer,108,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
5135,8286476,1,5,7,Cabeceras,27,CHOCÓ,31.65,1,Mujer,108,"Negro, mulato (afrodescendiente)",NaN,NaN,NaN,NaN,NaN,0.00
32904,8299104,1,3,7,Centros poblados y rural disperso,13,BOLÍVAR,128.58,1,Mujer,106,"Negro, mulato (afrodescendiente)",NaN,NaN,NaN,NaN,NaN,0.00
256662,8395585,1,3,10,Cabeceras,17,CALDAS,66.42,1,Mujer,106,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
396633,8468208,1,4,12,Centros poblados y rural disperso,25,CUNDINAMARCA,"1,231.96",1,Mujer,106,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
245532,8391286,1,5,10,Cabeceras,63,QUINDÍO,151.42,1,Mujer,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
145507,8348957,1,4,9,Cabeceras,50,META,241.71,1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
185317,8365558,1,3,9,Cabeceras,85,CASANARE,299.88,1,Mujer,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
366733,8446239,1,6,12,Cabeceras,08,ATLÁNTICO,"1,371.41",1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00


In [109]:
edad_min = df_empleo['edad'].min()
edad_max = df_empleo['edad'].max()

edad_min, edad_max

(0, 108)

La edad mínima de las personas que respondieron la encuesta es 0 años, mientras que la edad máxima es 108 años.

Escribe tu análisis aquí:


# Ejercicio 9: Groupby

Agrupar los datos por categorías, nos permite realizar análisis más profundos y obtener insights valiosos, que a simple vista no vemos en la base de datos.

1. Crea un nuevo dataframe llamado "df_sexo" donde se agrupe por 'sexo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de sexo en una columna.

In [110]:
### TU CÓDIGO AQUÍ ###
df_sexo = df_empleo.groupby('sexo')[['poblacion_total', 'ocupados']].sum()

In [111]:
df_sexo['porcentaje_ocupados'] = df_sexo['ocupados'] / df_sexo['poblacion_total']

In [112]:
df_sexo = df_sexo.reset_index()

In [113]:
df_sexo

,sexo,poblacion_total,ocupados,porcentaje_ocupados
0,Hombre,190530,"100,790.00",0.53
1,Mujer,218597,"81,063.00",0.37


2. Crea un nuevo dataframe llamado "df_educacion" donde se agrupe por 'nivel_educativo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de nivel educativo en una columna.

In [114]:
### TU CÓDIGO AQUÍ ###

df_educacion = df_empleo.groupby('sexo')[['poblacion_total', 'ocupados']].sum()

In [115]:
df_educacion['porcentaje_ocupados'] = df_educacion['ocupados'] / df_sexo['poblacion_total']

In [116]:
df_educacion = df_educacion.reset_index()

In [117]:
df_educacion

,sexo,poblacion_total,ocupados,porcentaje_ocupados
0,Hombre,190530,"100,790.00",NaN
1,Mujer,218597,"81,063.00",NaN


3. En una celda de Markdown escribe qué conclusiones puedes sacar de la información obtenida en df_sexo y df_educacion.

Escribe tu análisis aquí:

# Ejercicio 10: Filtrar datos

1. A partir de df_empleo, crea el DataFrame `df_ing` con observaciones donde `ingreso_laboral` no sea nulo y además sea mayor estrictamente al salario mínimo para 2026 (con subsidio de transporte).

In [118]:
### TU CÓDIGO AQUÍ ###
salario_minimo=2000000
df_ing = df_empleo[(df_empleo['ingreso_laboral'].notna())&(df_empleo['ingreso_laboral'] > salario_minimo)]

In [119]:
df_ing.reset_index(drop=True, inplace=True)
df_ing.head()

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
0,8283147,1,1,7,Centros poblados y rural disperso,08,ATLÁNTICO,57.99,1,Hombre,36,Ninguno de los anteriores,NaN,1.00,Asalariado,5022,Transporte fluvial de carga,"2,623,500.00"
1,8283163,1,1,7,Centros poblados y rural disperso,08,ATLÁNTICO,496.47,1,Hombre,39,Ninguno de los anteriores,Tecnológica,1.00,Asalariado,1103,"Producción de malta, elaboración de cervezas y...","3,000,000.00"
2,8283213,1,2,7,Cabeceras,66,RISARALDA,360.58,1,Hombre,24,Ninguno de los anteriores,Universitaria,1.00,Asalariado,8610,"Actividades de hospitales y clínicas, con inte...","2,700,000.00"
3,8283275,1,2,7,Cabeceras,66,RISARALDA,208.52,1,Mujer,35,Ninguno de los anteriores,Técnica profesional,1.00,Asalariado,4631,Comercio al por mayor de productos alimenticios,"2,200,000.00"
4,8283277,1,1,7,Centros poblados y rural disperso,66,RISARALDA,537.44,1,Hombre,69,Ninguno de los anteriores,Educación media,1.00,Asalariado,5611,Expendio a la mesa de comidas preparadas,"2,450,000.00"


2. ¿Cuántas personas tienen un salario estrictamente mayor al salario mínimo?

In [120]:
### TU CÓDIGO AQUÍ ###
len(df_ing)

38093

3. Crea el dataframe df_mujeres donde se encuentren solamente mujeres que están ubicadas en el departamento del Tolima, que son asalariadas y cuyo nivel educativo es educación media o superior.

In [121]:
### TU CÓDIGO AQUÍ ###
df_empleo['nivel_educativo'].value_counts()

,count
nivel_educativo,
Educación media,116635
Universitaria,32517
Técnica profesional,24503
Ninguno,12963
Tecnológica,10640
Especialización,7927
Maestría,3154
Doctorado,339
"No sabe, no informa",1


In [139]:
df_empleo['nivel_educativo'].unique()

[NaN, 'Educación media', 'Ninguno', 'Técnica profesional', 'Tecnológica', 'Universitaria', 'Maestría', 'Especialización', 'Doctorado', 'No sabe, no informa']
Categories (9, object): ['Doctorado', 'Educación media', 'Especialización', 'Maestría', ...,
                         'No sabe, no informa', 'Tecnológica', 'Técnica profesional', 'Universitaria']

In [125]:
df_empleo[df_empleo['sexo'] == 'Mujer']

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
1,8283143,1,2,7,Centros poblados y rural disperso,08,ATLÁNTICO,27.75,1,Mujer,34,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
3,8283145,1,2,7,Centros poblados y rural disperso,08,ATLÁNTICO,59.11,1,Mujer,50,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
7,8283147,1,2,7,Centros poblados y rural disperso,08,ATLÁNTICO,57.99,1,Mujer,37,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
8,8283147,1,3,7,Centros poblados y rural disperso,08,ATLÁNTICO,57.99,1,Mujer,10,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
9,8283147,1,4,7,Centros poblados y rural disperso,08,ATLÁNTICO,57.99,1,Mujer,21,Ninguno de los anteriores,Educación media,NaN,NaN,NaN,NaN,0.00
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409121,8473904,1,2,12,Cabeceras,91,AMAZONAS,64.83,1,Mujer,20,Ninguno de los anteriores,Educación media,NaN,NaN,NaN,NaN,0.00
409122,8473907,1,1,12,Centros poblados y rural disperso,73,TOLIMA,378.31,1,Mujer,61,Ninguno de los anteriores,NaN,1.00,Asalariado,9700,Actividades de los hogares individuales como e...,"1,400,000.00"
409124,8473908,1,2,12,Cabeceras,54,NORTE DE SANTANDER,534.55,1,Mujer,51,Ninguno de los anteriores,Maestría,1.00,Asalariado,8412,Actividades ejecutivas de la administración pú...,"7,956,000.00"
409125,8473908,1,3,12,Cabeceras,54,NORTE DE SANTANDER,534.55,1,Mujer,31,Ninguno de los anteriores,Universitaria,NaN,NaN,NaN,NaN,0.00


In [137]:
df_mujeres = df_empleo[
    (df_empleo['sexo'] == 'Mujer') &
    (df_empleo['nombre_dpto'] == 'TOLIMA') &
    (df_empleo['categoria_ocupacional'] == 'Asalariado') &
    (df_empleo['nivel_educativo'].notna()) &
    (df_empleo['nivel_educativo'].isin([
        'Educación media',
        'Técnica profesional',
        'Tecnológica',
        'Universitaria',
        'Especialización',
        'Maestría'
    ]))
]

4. ¿Cuáles son las dimensiones de df_mujeres?

In [138]:
### TU CÓDIGO AQUÍ ###
df_mujeres.shape

(1135, 18)